modules/gcp_gcs_bucket/inputs.tf

```diff
@@ -56,6 +56,24 @@ variable "lifecycle_rules" {
   default     = []
 }
 
+variable "object_versioning_enabled" {
+  type        = bool
+  description = "If set to true, the bucket will be versioned."
+  default     = true
+}
+
+variable "retention_lock_enabled" {
+  type        = bool
+  description = "If set to true, the bucket will be locked and objects in the bucket will be protected from deletion. Note that retention_policy cannot be used with object versioning. They are mutually exclusive."
+  default     = false
+}
+
+variable "retention_lock_duration_seconds" {
+  type        = number
+  description = "The duration in seconds that objects in the bucket must be retained and cannot be deleted or replaced. The value must be in between 604800(7 days) and 31536000(365 days)."
+  default     = 86400 # 1 day
```

Comment:
The line above says:

> The value must be in between 604800(7 days) and 31536000(365 days).

Reply:
https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules

Reply:
You can add a validation block so a plan will fail if you provide a value that is out-of-bounds, for example:

```hcl
variable "retention_lock_duration_seconds" {
  ...

  validation {
    condition     = var.retention_lock_enabled == true && (var. retention_lock_duration_seconds >= 604800 && var. retention_lock_duration_seconds <= 31536000)
    error_message = "The retention_lock_duration_seconds must be between 604800(7 days) and 31536000(365 days)."
  }
}
```